### ISSUE TITLE
Property updates don't trigger redraws on macOS 26 (Darwin 25.4) — regression between 1.15.1 and 1.16.0

### ISSUE THREAD

**Issue by cgommel** [slint-ui/slint#11472]
## Description

On macOS 26 ("Tahoe", Darwin 25.4.0), Slint **1.16.0** and current **master** (`582433b`, 1.17.0-dev) no longer redraw the window after property updates. The initial render works, and exactly one follow-up update propagates to the screen, after which the UI stays frozen even though the Slint reactivity system and all callbacks keep firing internally.

The issue is reproducible with a minimal Slint-only example — no Rust callbacks, no threads, no timers, no model data. A button that mutates an integer property inside a Slint `clicked` handler does not cause the dependent `Text` to re-render.

## Observed

- `cargo run` opens the window.
- Initial paint shows `c=0`.
- Clicking "inc" any number of times does **not** change the visible text — it stays `c=0` (or whatever value was shown right after the first redraw).
- Mouse hover highlights on the button still animate (Slint's internal hover binding works), but property-driven content does not.

Testing with a Rust-side `set_counter` via `Weak::upgrade` in a callback, `app.window().request_redraw()` after every change, and a 16 ms `slint::Timer` touching a dummy property — none of it makes additional redraws visible.

## Expected

`c` updates on every click and the `Text` re-renders accordingly.

## Version matrix on this Mac (macOS 26, Darwin 25.4.0, Apple Silicon)

| Slint version | Result |
|---|---|
| `=1.14.0` | works |
| `=1.15.1` | works |
| `=1.16.0` | **broken** |
| `git master` (`582433b`, 1.17.0-dev) | **broken** |

Backends tried on the broken versions:

- Default (winit-femtovg) — broken.
- `SLINT_BACKEND=winit-software` — broken.
- `winit-skia` via `features = ["renderer-skia"]` with `SLINT_BACKEND=winit-skia` — broken.
- `default-features = false` + `features = ["std", "compat-1-2", "backend-winit", "renderer-skia"]` (no `accessibility`) — broken.

So the renderer doesn't matter; it's somewhere at the event-loop or dirty-tracking layer.

## Workaround

Pinning to `slint = "=1.15.1"` is working fine for us. Happy to provide additional bisect data or run extra diagnostics if helpful.

### Reproducible Code (if applicable)

```slint
`Cargo.toml`:

[package]
name = "slint-redraw-repro"
version = "0.1.0"
edition = "2021"

[dependencies]
slint = "=1.16.0"   # also broken on git master / 1.17


`src/main.rs`:


slint::slint! {
    import { Button } from "std-widgets.slint";

    export component PureWindow inherits Window {
        width: 400px;
        height: 200px;
        title: "Slint Pure";
        in-out property <int> c: 0;

        VerticalLayout {
            padding: 20px;
            spacing: 12px;
            Text {
                text: "c=" + c;
                font-size: 32px;
                horizontal-alignment: center;
            }
            Button {
                text: "inc";
                clicked => { c += 1; }
            }
        }
    }
}

fn main() -> Result<(), Box<dyn std::error::Error>> {
    slint::ComponentHandle::run(&PureWindow::new()?)?;
    Ok(())
}
```

## Environment

- macOS: 26 ("Tahoe"), Darwin 25.4.0
- Architecture: aarch64 (Apple Silicon)
- Rust: `rustc 1.95.0 (59807616e 2026-04-14)` (stable)
- winit: `0.30.13` (as transitively pinned by Slint 1.16)

**Comment by tronical** [comment]
Ouch, this could be a regression from my display link change. I'll debug this tomorrow. Thanks for the report!

**Comment by tronical** [comment]
Ok, this is odd. I just tried your example and it works for me. I'm also on apple silicon, macOS 26.4.1. 

Could you try something for me and in `internal/backends/winit/frame_throttle/macos_display_link.rs` make `try_create` fail unconditionally?

**Comment by mattly** [comment]
for what it's worth, I'm seeing this for all the examples on `master` and `v0.16.0` on macOS 15.7; they work for wasm but not for macOS binaries

**Comment by tronical** [comment]
Any help here would be greatly appreciated in debugging. Do you see this with any style? Does it somehow related to animations?

**Comment by mattly** [comment]
> Could you try something for me and in `internal/backends/winit/frame_throttle/macos_display_link.rs` make `try_create` fail unconditionally?

I inserted `return None;` into the first line of this function, and the examples work as macOS binaries.

I am unfamiliar with Slint in particular; I'm evaluating it alongside a few other cross-platform UI toolkits, but I'm not really sure how to manipulate the other factors like style or animations

**Comment by tronical** [comment]
I'd be super grateful if somebody could give https://github.com/slint-ui/slint/pull/11476 a try

**Comment by tronical** [comment]
Fixed with commit 5fafdaa
